client/src/status_bar.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { DenoExtensionContext } from "./types";
+
+import * as vscode from "vscode";
+
+export class DenoStatusBar {
+  readonly #inner: vscode.StatusBarItem;
+
+  constructor() {
+    this.#inner = vscode.window.createStatusBarItem(
+      vscode.StatusBarAlignment.Right,
+      0,
+    );
+  }
+
+  dispose() {
+    this.#inner.dispose();
+  }
+
+  refresh(extensionContext: DenoExtensionContext) {
```

Comment:
I kind of prefer a single state tree then parts of the application listen for state changes and update accordingly. That would be a big refactor though and not sure if desired.